data_steward/gcloud/bq/__init__.py

```diff
@@ -227,20 +229,33 @@ def define_dataset(self, dataset_id: str, description: str,
 
         return dataset
 
-    def copy_dataset(self, input_dataset: str, output_dataset: str) -> list:
+    def copy_dataset(self,
+                     input_dataset: str,
+                     output_dataset: str,
+                     job_config: CopyJobConfig = None,
+                     include_metadata: bool = False) -> list:
         """
         Copies tables from source dataset to a destination datasets
 
         :param input_dataset: fully qualified name of the input(source) dataset
         :param output_dataset: fully qualified name of the output(destination) dataset
         :return: incomplete jobs
         """
+        if include_metadata and not job_config:
+            job_config = CopyJobConfig()
+            
         # Copy input dataset tables to backup and staging datasets
         tables = super(BigQueryClient, self).list_tables(input_dataset)
         job_list = []
         for table in tables:
             staging_table = f'{output_dataset}.{table.table_id}'
-            job = self.copy_table(table, staging_table)
+            if include_metadata:
+                job_config.labels = {
+                    'table_name': table.table_id,
+                    'copy_from': input_dataset,
+                    'copy_to': output_dataset
+                }
+            job = self.copy_table(table, staging_table, job_config=job_config)
```

Comment:
If we're going to add labels, lets add them for all copy jobs.  This would become
```python
            job_config = CopyJobConfig()
            job_config.labels = {
                'table_name': table.table_id,
                'copy_from': input_dataset,
                'copy_to': output_dataset
            }
            job = self.copy_table(table, staging_table, job_config=job_config)
```